packages/typespec-azure-resource-manager/src/rules/arm-no-record.ts

```diff
@@ -0,0 +1,47 @@
+import { Model, SemanticNodeListener, createRule } from "@typespec/compiler";
+
+function isRecordType(model: Model): boolean {
+  return model.name === "Record";
+}
+
+export const armNoRecordRule = createRule({
+  name: "arm-no-record",
+  severity: "warning",
+  description: "Don't use Record types for ARM resources.",
+  url: "https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/rules/no-record",
+  messages: {
+    default:
+      "Definitions must not be of type Record. ARM requires Resource provider teams to define types explicitly.",
```

Comment:
```suggestion
      "Properties must not be of type Record. ARM requires Resource provider teams to define types explicitly.",
```
as this is currently where its used for(though maybe changing the key to be explicit)

Reply:
I would make this 'Model Properties and Parameters', as I think this is actually what will be flagged by the rule.